File: backend/src/modules/lcr/lcr.controller.ts
import { Controller, Get, Post, Body, Patch, Param, Delete, Query, ParseIntPipe } from '@nestjs/common';
import { ApiTags, ApiOperation } from '@nestjs/swagger';
import { LcrService } from './lcr.service';
import { PaginationDto } from '../../common/dto/pagination.dto';
import { ApiResponseDto } from '../../common/dto/api-response.dto';

@ApiTags('LCR 路由')
@Controller('api/lcr')
export class LcrController {
  constructor(private readonly lcrService: LcrService) {}

  // Gateway
  @Get('gateways')
  @ApiOperation({ summary: '获取网关列表' })
  async findAllGateways(@Query() dto: PaginationDto) {
    return ApiResponseDto.success(await this.lcrService.findAllGateways(dto));
  }

  @Post('gateways')
  async createGateway(@Body() data: any) {
    return ApiResponseDto.success(await this.lcrService.createGateway(data), '创建成功');
  }

  @Patch('gateways/:id')
  async updateGateway(@Param('id', ParseIntPipe) id: number, @Body() data: any) {
    return ApiResponseDto.success(await this.lcrService.updateGateway(id, data), '更新成功');
  }

  @Delete('gateways/:id')
  async removeGateway(@Param('id', ParseIntPipe) id: number) {
    await this.lcrService.removeGateway(id);
    return ApiResponseDto.success(null, '删除成功');
  }

  // Rule
  @Get('rules')
  @ApiOperation({ summary: '获取规则列表' })
  async findAllRules(@Query() dto: PaginationDto) {
    return ApiResponseDto.success(await this.lcrService.findAllRules(dto));
  }

  @Post('rules')
  async createRule(@Body() data: any) {
    return ApiResponseDto.success(await this.lcrService.createRule(data), '创建成功');
  }

  @Patch('rules/:id')
  async updateRule(@Param('id', ParseIntPipe) id: number, @Body() data: any) {
    return ApiResponseDto.success(await this.lcrService.updateRule(id, data), '更新成功');
  }

  @Delete('rules/:id')
  async removeRule(@Param('id', ParseIntPipe) id: number) {
    await this.lcrService.removeRule(id);
    return ApiResponseDto.success(null, '删除成功');
  }

  // Target
  @Get('targets')
  @ApiOperation({ summary: '获取目标列表' })
  async findAllTargets(@Query() dto: PaginationDto) {
    return ApiResponseDto.success(await this.lcrService.findAllTargets(dto));
  }

  @Post('targets')
  async createTarget(@Body() data: any) {
    return ApiResponseDto.success(await this.lcrService.createTarget(data), '创建成功');
  }

  @Patch('targets/:id')
  async updateTarget(@Param('id', ParseIntPipe) id: number, @Body() data: any) {
    return ApiResponseDto.success(await this.lcrService.updateTarget(id, data), '更新成功');
  }

  @Delete('targets/:id')
  async removeTarget(@Param('id', ParseIntPipe) id: number) {
    await this.lcrService.removeTarget(id);
    return ApiResponseDto.success(null, '删除成功');
  }

  // RPC
  @Post('reload')
  @ApiOperation({ summary: '重载 LCR' })
  async reload() {
    await this.lcrService.reload();
    return ApiResponseDto.success(null, '重载成功');
  }

  @Get('dump-gws')
  @ApiOperation({ summary: '导出网关' })
  async dumpGws() {
    return ApiResponseDto.success(await this.lcrService.dumpGws());
  }

  @Get('dump-rules')
  @ApiOperation({ summary: '导出规则' })
  async dumpRules() {
    return ApiResponseDto.success(await this.lcrService.dumpRules());
  }
}

File: kam-admin-module-routing/src/main/java/io/kamailio/admin/modules/lcr/controller/LcrController.java
package io.kamailio.admin.modules.lcr.controller;

import io.kamailio.admin.common.dto.ApiResponse;
import io.kamailio.admin.common.dto.PaginatedResult;
import io.kamailio.admin.common.dto.PaginationDto;
import io.kamailio.admin.modules.lcr.entity.LcrGw;
import io.kamailio.admin.modules.lcr.entity.LcrRule;
import io.kamailio.admin.modules.lcr.entity.LcrRuleTarget;
import io.kamailio.admin.modules.lcr.service.LcrService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@Tag(name = "LCR 路由")
@RestController
@RequestMapping("/api/lcr")
public class LcrController {
    private final LcrService service;

    public LcrController(LcrService service) { this.service = service; }

    @GetMapping("/gateways")
    @Operation(summary = "获取网关列表")
    public ResponseEntity<ApiResponse<PaginatedResult<LcrGw>>> findAllGateways(PaginationDto dto) {
        return ResponseEntity.ok(ApiResponse.success(service.findAllGateways(dto)));
    }

    @PostMapping("/gateways")
    public ResponseEntity<ApiResponse<LcrGw>> createGateway(@RequestBody LcrGw data) {
        return ResponseEntity.ok(ApiResponse.success(service.createGateway(data), "创建成功"));
    }

    @PatchMapping("/gateways/{id}")
    public ResponseEntity<ApiResponse<LcrGw>> updateGateway(@PathVariable Integer id, @RequestBody LcrGw data) {
        return ResponseEntity.ok(ApiResponse.success(service.updateGateway(id, data), "更新成功"));
    }

    @DeleteMapping("/gateways/{id}")
    public ResponseEntity<ApiResponse<Void>> removeGateway(@PathVariable Integer id) {
        service.removeGateway(id);
        return ResponseEntity.ok(ApiResponse.success(null, "删除成功"));
    }

    @GetMapping("/rules")
    @Operation(summary = "获取规则列表")
    public ResponseEntity<ApiResponse<PaginatedResult<LcrRule>>> findAllRules(PaginationDto dto) {
        return ResponseEntity.ok(ApiResponse.success(service.findAllRules(dto)));
    }

    @PostMapping("/rules")
    public ResponseEntity<ApiResponse<LcrRule>> createRule(@RequestBody LcrRule data) {
        return ResponseEntity.ok(ApiResponse.success(service.createRule(data), "创建成功"));
    }

    @PatchMapping("/rules/{id}")
    public ResponseEntity<ApiResponse<LcrRule>> updateRule(@PathVariable Integer id, @RequestBody LcrRule data) {
        return ResponseEntity.ok(ApiResponse.success(service.updateRule(id, data), "更新成功"));
    }

    @DeleteMapping("/rules/{id}")
    public ResponseEntity<ApiResponse<Void>> removeRule(@PathVariable Integer id) {
        service.removeRule(id);
        return ResponseEntity.ok(ApiResponse.success(null, "删除成功"));
    }

    @GetMapping("/targets")
    @Operation(summary = "获取目标列表")
    public ResponseEntity<ApiResponse<PaginatedResult<LcrRuleTarget>>> findAllTargets(PaginationDto dto) {
        return ResponseEntity.ok(ApiResponse.success(service.findAllTargets(dto)));
    }

    @PostMapping("/targets")
    public ResponseEntity<ApiResponse<LcrRuleTarget>> createTarget(@RequestBody LcrRuleTarget data) {
        return ResponseEntity.ok(ApiResponse.success(service.createTarget(data), "创建成功"));
    }

    @PatchMapping("/targets/{id}")
    public ResponseEntity<ApiResponse<LcrRuleTarget>> updateTarget(@PathVariable Integer id, @RequestBody LcrRuleTarget data) {
        return ResponseEntity.ok(ApiResponse.success(service.updateTarget(id, data), "更新成功"));
    }

    @DeleteMapping("/targets/{id}")
    public ResponseEntity<ApiResponse<Void>> removeTarget(@PathVariable Integer id) {
        service.removeTarget(id);
        return ResponseEntity.ok(ApiResponse.success(null, "删除成功"));
    }

    @PostMapping("/reload")
    @Operation(summary = "重载 LCR")
    public ResponseEntity<ApiResponse<Void>> reload() {
        service.reload();
        return ResponseEntity.ok(ApiResponse.success(null, "重载成功"));
    }

    @GetMapping("/dump-gws")
    @Operation(summary = "导出网关")
    public ResponseEntity<ApiResponse<Object>> dumpGws() {
        return ResponseEntity.ok(ApiResponse.success(service.dumpGws()));
    }

    @GetMapping("/dump-rules")
    @Operation(summary = "导出规则")
    public ResponseEntity<ApiResponse<Object>> dumpRules() {
        return ResponseEntity.ok(ApiResponse.success(service.dumpRules()));
    }
}
